### PR TITLE
Wr 295 access control policy endpoints

### PR DESCRIPTION
**Pull request by rosasck**
Keycloak endpoints for enable/disable access control (yep! theres a total of 5 keycloak endpoints for that)

and client side endpoints for 
* enable/disable access control 
* Add groups to policy 
* remove groups from policy

### CODE REVIEW

File: identityClient/api.go [hunk 1]
```diff
@@ -1190,3 +1190,30 @@ type FederatedIdentityKeyCheckRequest struct {
 	ClientID                        uuid.UUID `json:"client_id"`
 	PasswordDerivedPublicSigningKey string    `json:"password_derived_public_signing_key"`
 }
+
+// AccessControlPolicyRequest
```
Comment (rosasck) [inline identityClient/api.go:1194]: need to add more context

File: keycloakClient/api.go [hunk 2]
```diff
@@ -533,3 +533,31 @@ type InitiatePKCELogin struct {
 	CodeChallenge       string `schema:"code_challenge"`
 	CodeChallengeMethod string `schema:"code_challenge_method"`
 }
+
+// CreateResourcePermissionRequest
```
Comment (rosasck) [inline keycloakClient/api.go:537]: need to add more context

File: identityClient/api.go [hunk 3]
```diff
@@ -1190,3 +1190,30 @@ type FederatedIdentityKeyCheckRequest struct {
...
+}
+
+// AccessControlPolicyGroup
+type AccessControlPolicyGroup struct {
+	ID               string `json:"id"`
+	ExtendToChildren bool   `json:"extend_to_children"`
+}
+
+// UnenrollAccessControlPolicy
+type UnenrollAccessControlPolicy struct {
```
Comment (galxy25) [inline identityClient/api.go:1215]: Good naming, but prefer add/remove terminology for the sake of simplicity unless there is some additional action that happens when these policies are set up aside from updating who can access an application

File: identityClient/api.go [hunk 4]
```diff
@@ -1190,3 +1190,30 @@ type FederatedIdentityKeyCheckRequest struct {
...
+
+// AccessControlPolicyRequest
+type AccessControlPolicyRequest struct {
+	RealmName     string
+	ApplicationID string
+	Enable        bool `json:"enable"`
+}
+
+// EnrollAccessControlPolicy
+type EnrollAccessControlPolicy struct {
```
Comment (galxy25) [inline identityClient/api.go:1202]: `Request` postfix for this and others where appropriate please

File: keycloakClient/keycloakClient.go [hunk 5]
```diff
@@ -1313,3 +1320,39 @@ func (c *Client) InitiateLogin(realmName string, loginURLEncoded InitiatePKCELog
 	}
 	return resp.RawResponse, nil
 }
+
+// Access Control Policy Endpoints
```
Comment (galxy25) [inline keycloakClient/keycloakClient.go:1324]: if ease of navigation / separation of concerns is the objective of this comment, consider moving these methods into a `client_applications.go` file

**Review by galxy25 — APPROVED** [review]

File: keycloakClient/keycloakClient.go [hunk 6]
```diff
@@ -535,96 +542,6 @@ func (c *Client) GetRealm(accessToken string, realmName string) (RealmRepresenta
 	return response, err
 }
 
-// GetClient get the representation of the client. idClient is the id of client (not client-id).
```
Comment (rosasck) [inline keycloakClient/keycloakClient.go:538]: Since i made a client applications file, I transferred over the existing methods.

**Review by efabens — APPROVED** [review]

**Pull request closed without merging**